habitat_baselines/common/rollout_storage.py

```diff
@@ -208,13 +219,37 @@ def recurrent_generator(
                     num_environments, num_mini_batch
                 )
             )
+
+        dones_cpu = (
+            torch.logical_not(self.buffers["masks"])
+            .cpu()
+            .view(-1, self._num_envs)
+            .numpy()
+        )
         for inds in torch.randperm(num_environments).chunk(num_mini_batch):
-            batch = self.buffers[0 : self.current_rollout_step_idx, inds]
-            batch["advantages"] = advantages[
-                0 : self.current_rollout_step_idx, inds
-            ]
+            curr_slice = (slice(0, self.current_rollout_step_idx), inds)
+
+            batch = self.buffers[curr_slice]
+            if advantages is not None:
+                batch["advantages"] = advantages[curr_slice]
             batch["recurrent_hidden_states"] = batch[
                 "recurrent_hidden_states"
             ][0:1]
 
-            yield batch.map(lambda v: v.flatten(0, 1))
+            batch.map_in_place(lambda v: v.flatten(0, 1))
+
+            batch["rnn_build_seq_info"] = build_rnn_build_seq_info(
+                device=self.device,
+                build_fn_result=build_pack_info_from_dones(
+                    dones_cpu[curr_slice],
+                ),
+            )
+
+            yield batch
+
+    def __getstate__(self):
+        return self.__dict__.copy()
```

Comment:
Yikes, why a copy? Even if it's shallow?

Reply:
The rollout storage for VER inherits from this and it has to delete stuff from the state dict, but it should probably be the one to make the copy.